memgpt/cli/cli_config.py

```diff
@@ -96,16 +97,32 @@ def configure_llm_endpoint(config: MemGPTConfig):
     return model_endpoint_type, model_endpoint
 
 
-def configure_model(config: MemGPTConfig, model_endpoint_type: str):
+def configure_model(config: MemGPTConfig, model_endpoint_type: str, model_endpoint: str):
     # set: model, model_wrapper
     model, model_wrapper = None, None
     if model_endpoint_type == "openai" or model_endpoint_type == "azure":
-        model_options = ["gpt-4", "gpt-4-1106-preview", "gpt-3.5-turbo", "gpt-3.5-turbo-16k"]
-        # TODO: select
+        try:
+            if model_endpoint_type == "openai":
+                model_options = openai_get_model_list(url=model_endpoint, api_key=config.openai_key)
+            elif model_endpoint_type == "azure":
+                model_options = azure_openai_get_model_list(url=model_endpoint, api_key=config.azure_key, api_version=config.azure_version)
```

Comment:
This is potentially problematic because `config.azure_key` and `config.azure_version` may not have been defined at this point?

Reply:
I think this might error on the first run of memgpt configure, if you don't already have credentials put in?

Reply:
This PR should mean this won't happen since it forces keys to be specified up front: https://github.com/cpacker/MemGPT/pull/675

Reply:
Actually it meant it for openai, but not for azure, which I patched here https://github.com/cpacker/MemGPT/pull/630/commits/4d38c14defc0dc5ab924f239f1450aca99dd67ec